src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,261 @@
+        User actual = registServImpl.register(expected);
+        assertEquals(expected,
+                actual,
+                "Registration should succeed when the password length is 6.");
+    }
+
+    @Test
+    public void register_8_password_length_Ok() {
+        expected.setPassword(OK_PASSWORD_EIGHT_LENGTH);
+        expected.setAge(OK_AGE_TWENTY_TWO);
+        expected.setLogin(OK_LOGIN_TEN_LENGTH);
+        User actual = registServImpl.register(expected);
+        assertEquals(expected,
+                actual,
+                "Registration should succeed when the password length is 8.");
+    }
+
+    @Test
+    public void register_0_login_length_notOk() {
+        expected.setPassword(OK_PASSWORD_TEN_LENGTH);
+        expected.setAge(OK_AGE_TWENTY_TWO);
+        expected.setLogin(NOT_OK_LOGIN_ZERO_LENGTH);
+        assertThrows(RegistrationException.class,
+                () -> registServImpl.register(expected),
+                "Registration should throw an exception when the login length is 0.");
+    }
+
+    @Test
+    public void register_3_login_length_notOk() {
+        expected.setPassword(OK_PASSWORD_TEN_LENGTH);
+        expected.setAge(OK_AGE_TWENTY_TWO);
+        expected.setLogin(NOT_OK_LOGIN_THREE_LENGTH);
+        assertThrows(RegistrationException.class,
+                () -> registServImpl.register(expected),
+                "Registration should throw an exception when the login length is 3.");
+    }
+
+    @Test
+    public void register_5_login_length_notOk() {
+        expected.setPassword(OK_PASSWORD_TEN_LENGTH);
+        expected.setAge(OK_AGE_TWENTY_TWO);
+        expected.setLogin(NOT_OK_LOGIN_FIVE_LENGTH);
+        assertThrows(RegistrationException.class,
+                () -> registServImpl.register(expected),
+                "Registration should throw an exception when the login length is 5.");
+    }
+
+    @Test
+    public void register_6_login_length_Ok_edge_case() {
+        expected.setPassword(OK_PASSWORD_TEN_LENGTH);
+        expected.setAge(OK_AGE_TWENTY_TWO);
+        expected.setLogin(NOT_OK_LOGIN_SIX_LENGTH);
+        User actual = registServImpl.register(expected);
+        assertEquals(expected,
+                actual,
+                "Registration should succeed when the login length is 6.");
+    }
+
+    @Test
+    public void register_8_login_length_Ok() {
+        expected.setPassword(OK_PASSWORD_TEN_LENGTH);
+        expected.setAge(OK_AGE_TWENTY_TWO);
+        expected.setLogin(OK_PASSWORD_EIGHT_LENGTH);
+        User actual = registServImpl.register(expected);
+        assertEquals(expected,
+                actual,
+                "Registration should succeed when the login length is 8.");
+    }
+
+    @Test
+    public void register_user_is_already_registered_notOk() {
+        expected.setPassword(OK_PASSWORD_TEN_LENGTH);
+        expected.setAge(OK_AGE_TWENTY_TWO);
+        expected.setLogin(OK_PASSWORD_EIGHT_LENGTH);
+        registServImpl.register(expected);
```

Comment:
Here we should rather directly pre-save user with storage. The reason for that is that here out registration service is under test, so we cannot rely in it's behaviour